### PR TITLE
Document support for thread-local storage within enclaves.

### LINKED ISSUE #1536: Thread Locals: threading-model and assertion
This issue captures the final tweaks needed to support thread-local storage in 0.5x

- Decide whether to specify the threading-model as part of compiler options. If so, add it to pkgconfig.
   Choose the correct threading model, inital-exec or local-exec.
   If not specified as compiler options, then document the options if the end-user gets a link error with __tls_get_addr

- Investigate and remove the assertion put in place to detect when thread-locals information in ELF section and program header differ

### PR DESCRIPTION
**Pull request by anakrish**
Fixes #1536

### CODE REVIEW

File: docs/ThreadLocalSupport.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+Thread-Local Storage Support In Enclaves
+----------------------------------------
+
+Thread-locals are supported within enclaves since release 0.5.
+
```
Comment (soccerGB) [inline docs/ThreadLocalSupport.md]: release 0.5. --> release 0.5.0

File: docs/ThreadLocalSupport.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+When the _ecall_ returns, the enclave-thread terminates and the thread-locals are cleaned up.
+Cleaning up thread-locals may involve calling destructors.
+
+Open Enclave SDK requires that the enclaves be compiled using the **_local-exec_** thread-local model via
+**_-ftls-model=local-exec_** GCC/Clang option. The **_local-exec_** thread-local model results in the most
+optimal thread-local implementation for enclaves.
+
+See [GCC Code Gen Options](https://gcc.gnu.org/onlinedocs/gcc/Code-Gen-Options.html) for -ftls-model options.
+
+See [ELF Handling For Thread-Local Storage](https://www.akkadia.org/drepper/tls.pdf) for a detailed explanation of various thread-local storage models.
```
Comment (soccerGB) [inline docs/ThreadLocalSupport.md]: This is a nice link to the TLS information.

File: docs/ThreadLocalSupport.md (hunk 1 above)
Comment (anakrish) [reply docs/ThreadLocalSupport.md]: Fixed.

File: docs/ThreadLocalSupport.md (hunk 2 above)
Comment (anakrish) [reply docs/ThreadLocalSupport.md]: Thanks.

**Pull request merged**